### PR TITLE
GSoC18

### PR DESCRIPTION
**Pull request by sbansal3096**

### CODE REVIEW

File: app/src/main.rs [hunk 1]
```diff
@@ -10,6 +10,8 @@ fn main() {
     m.log_violation("unauthorized");
     let violations = m.take_violations();
     for v in violations {
+        //let d1:<U:=
+        //let d=v.as_ref();
         println!("{}", v.as_ref());
```
Comment (sbansal3096) [inline app/src/main.rs:15]: @jdm 
I am really stuck at this error in this line. Could you please help?
_________________________________________________________________
 the type of this value must be known in this context
  --> src/main.rs:15:24
   |
15 |         println!("{}", v.as_ref());
   |                        ^^^^^^^^^^


_________________________________________________________________
I have used traits on properties of Logger and LogEntry to remove the dependancy of base on middle crate. Is my approach right or it needs some improvement?

Comment (jdm) [reply app/src/main.rs:15]: Try this:
```rust
let s: &str = v.as_ref();
println!("{}", s);
```

File: app/src/main.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 extern crate base;
 extern crate middle;
 
-use base::do_a_bunch_of_things;
+use base::{do_a_bunch_of_things,AsRef,as_ref};
```
Comment (jdm) [inline app/src/main.rs:4]: This change may also be confusing the compiler, because it's not a valid import.

Comment (sbansal3096) [reply app/src/main.rs:4]: oops sorry, well i have also tried wthout this change. Actually I made this change only while debugging the error.

File: app/src/main.rs (hunk 1 above)
Comment (sbansal3096) [reply app/src/main.rs:15]: Yes I have tried this, still the error remains same.

Comment (jdm) [reply app/src/main.rs:15]: Oh, the problem is that you're calling create_middleware but not providing any concrete types (like `create_middleware::<Foo, Bar>()`), so it's trying to infer them and getting stuck.

Comment (sbansal3096) [reply app/src/main.rs:15]: Oh yes! That was it. Thank you so much for helping me.

**Pull request closed without merging**